flybywiresim/aircraft
# refactor(efb): check simbrief a/c type against build

Fixes #[issue_no]

## Summary of Changes

- Check that imported simbrief profile matches the currently selected aircraft type
- If incorrect A/C type is detected, import the simbrief data, but create a toast notification and warn about the detection of the invalid aircraft type.

## Screenshots (if necessary)


![image](https://github.com/flybywiresim/aircraft/assets/15316958/bd919561-125b-41b6-9dd1-7652cf8c5078)


## References





## Additional context



Discord username (if different from GitHub): 2cas

## Testing instructions

1. Make a *correct* Simbrief flight and generate OFP (i.e. with the correct A20N A320neo profile) 
1. in flyPad EFB home page, import the simbrief data (Import Simbrief Data button)
1. Try to import this new flight plan, no warning should be displayed. (Other than fuel and payload loaded which is normal)

![image](https://github.com/flybywiresim/aircraft/assets/15316958/1f5986a6-999e-450b-ac21-83985721422b)

1. In Simbrief, generate a new OFP with an incorrect aircraft type (A330, ATR72, etc) and generate ofp
1. Simbrief data should still import, with the flight widget displaying flight/departure/arrival information and be usable in the payload/fuel pages however as before.
1. FlyPad should otherwise function normally, other than this new notification.

3. A notification should be displayed after the import that warns about the incorrect A/C type.


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page
